### PR TITLE
Issue 20 - Item List

### PR DESCRIPTION
**Pull request by sylochloe**
### Pull Request Summary

- Create itemTable component to display all the items in a table
- Add a fetch call in itemTable.js to get all the items from datastore

The table with test data:
Screen Shot 2023-04-27 at 9 45 25 PM

REMARKS: I removed the catergorys field from the table in the existing code, so when you can fetch all the items, the table won't crash. We can add it back and fix it once the call is working.

### Pull Request Checklist

- [x] Code is neat, readable, and works
- [x] Comments are appropriate
- [x] The commit messages follows our [guidelines](https://h4i.notion.site/Conventional-Commits-593452ad1179489399ad3bd696ef772a)
- [x] The developer name is specified
- [x] The summary is completed
- [x] Ask for a review in communication channels

### Reviewer Checklist - PULL REQUEST REVIEWER ONLY

IMPORTANT: The rest of the sections in this checklist should only be filled out by authorized pull request reviewers. If you are the individual contributor, do not fill out the rest of the fields or check the boxes.

## Reviewer: Full Name

NOTE: Pull requests should only be merged when all boxes are checked.

- [ ] Assign yourself as reviewer if not assigned
- [ ] The reviewer name is specified
- [ ] The code is fully reviewed
- [ ] Meaningful feedback is given
- [ ] Let developer know a review has been made

### CODE REVIEW

File: src/components/ItemTable.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    updatedAt: "2023-03-13T01:32:37.785Z",
+  },
+];
+
+const columns: GridColDef[] = [
+  { field: "name", headerName: "Name", minWidth: 200 },
+  { field: "value", headerName: "Price", minWidth: 150 },
+  { field: "dataAdded", headerName: "Date Added", minWidth: 150 },
+  { field: "createdAt", headerName: "Created At", minWidth: 200 },
+  { field: "updatedAt", headerName: "Updated At", minWidth: 200 },
```
Comment (jlaksana) [inline src/components/ItemTable.tsx:38]: Let's not show the updatedAt field. Also let's combine dateAdded with createdAt. createdAt is given by default by Amplify so we can just remove this field to use amplify's field.

File: src/components/ItemTable.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  // TODO: put the Categorys back once the fetch call is fixed
```
Comment (jlaksana) [inline src/components/ItemTable.tsx:39]: We will have to update this relationship to be in proper english. Thanks for finding this

File: src/components/ItemTable.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        console.log("Error retrieving posts", error);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  return (
+    <div style={{ height: 300, width: "100%" }}>
+      <DataGrid
```
Comment (jlaksana) [inline src/components/ItemTable.tsx:75]: Currently the pagination rows per page is 100. Let's limit it to 25 for now.

File: src/components/ItemTable.tsx (hunk 1 above)
Comment (sylochloe) [reply src/components/ItemTable.tsx:38]: Sounds good. Will remove it from the table. Thanks

File: src/components/ItemTable.tsx (hunk 3 above)
Comment (sylochloe) [reply src/components/ItemTable.tsx:73]: If I set it to 25 as the fixed PageSize, the table looks a bit weird. Therefore, I set 3 options for the pageSize (10, 25, 50), the default size is 25.

**Review by jlaksana — APPROVED** [review]
Looks great, thanks for taking this on!

**Pull request merged**